validator/api/src/main/java/tech/pegasys/artemis/validator/api/ValidatorApiChannel.java

```diff
@@ -41,7 +42,9 @@
 
   SafeFuture<Optional<Attestation>> createAggregate(AttestationData attestationData);
 
-  void subscribeToBeaconCommittee(int committeeIndex, UnsignedLong aggregationSlot);
+  void subscribeToBeaconCommitteeForAggregation(int committeeIndex, UnsignedLong aggregationSlot);
+
+  void updateRandomSubnetSubscriptions(Map<Integer, UnsignedLong> subnetIdToUnsubscriptionSlot);
```

Comment:
I'd define a small class that holds the subnet id & duration of the subscription:
```suggestion
  void updatePersistentSubnetSubscriptions(List<SubnetSubscription> subscriptions);
```

Reply:
yea that's a better name.

Reply:
Done.